q2cli/click/type.py

```diff
@@ -192,7 +192,15 @@ def _convert_metadata(self, value, param, ctx):
     def _convert_primitive(self, value, param, ctx):
         import qiime2.sdk.util
 
-        return qiime2.sdk.util.parse_primitive(self.type_expr, value)
+        try:
+            return qiime2.sdk.util.parse_primitive(self.type_expr, value)
+        except ValueError:
+            args = ', '.join(map(repr, value))
+            expr = qiime2.sdk.util.type_from_ast(self.type_ast)
+            raise click.BadParameter(
+                'recieved <%s> as an argument, which is incompatible'
```

Comment:
Drive-by comment: 
```suggestion
                'received <%s> as an argument, which is incompatible'
```

Reply:
It's misspelled all over the place and I copy-pasted :laughing: I'll fix them all